core/math/math.h

```diff
@@ -240,29 +246,40 @@ namespace MR
   }
 
   //! write the transform \a M to file
-  inline void save_transform (const transform_type& M, const std::string& filename, const bool write_cmdstring = false)
+  inline void save_transform (const transform_type& M, const std::string& filename, const vector<std::string>& header_lines)
```

Comment:
I reckon I'd be in favour of merging both aspects. This modification forces the user to format the command history string as soon as they want to add any comments, which I think is going to lead to inconsistencies down the track. Why not have a declaration with both the `vector<string>` and `bool` arguments, like this:
```
void save_matrix (const MatrixType& M, const std::string& filename, const vector<std::string> header_lines = vector<std::string>(), const bool write_cmdstring = true)
```
Note this assumes that we do want to default to inserting the command history every time - which I'd be happy to support. 

Another potential modification would be to add the version string in there along with the command history, if that bool is true. Thoughts?

Reply:
Part of the reasoning there was that if it's just the one line in isolation, you can just add the command string as a comment and that's it; whereas if you're wanting to have more than that, there's a decent chance that you may want the comments in some kind of format (e.g. key-value pairs), and therefore would do it all manually.

I'm happy to go with that solution, but I think the command string would need to be prepended with "`# command_history:`"...?

> Another potential modification would be to add the version string in there along with the command history

Already done: the variable "command_string" has the command, double-space, then the software version in brackets, just as has always be done in `command_history`. So maybe "`command_string`" isn't the ideal name?

Reply:
See 5eec075.